IdentityCore/src/throttling/model/MSIDThrottlingModelFactory.m

```diff
@@ -0,0 +1,134 @@
+    if(throttleType == MSIDThrottlingTypeNone) return nil;
+    return [self generateModelFromErrorResponse:errorResponse
+                                        request:request
+                                   throttleType:throttleType
+                                    cacheRecord:nil
+                                    accessGroup:accessGroup];
+}
+
++ (MSIDThrottlingModelBase *)generateModelFromErrorResponse:(NSError * _Nullable)errorResponse
+                                                    request:(id<MSIDThumbprintCalculatable>)request
+                                               throttleType:(MSIDThrottlingType)throttleType
+                                                cacheRecord:(MSIDThrottlingCacheRecord * _Nullable)cacheRecord
+                                                accessGroup:(NSString *)accessGroup
+{
+    if(throttleType == MSIDThrottlingType429)
+    {
+        return [[MSIDThrottlingModel429 alloc] initWithRequest:request cacheRecord:cacheRecord errorResponse:errorResponse accessGroup:accessGroup];
+    }
+    else
+    {
+        return [[MSIDThrottlingModelInteractionRequire alloc] initWithRequest:request cacheRecord:cacheRecord errorResponse:errorResponse accessGroup:accessGroup];
+    }
+}
+
++ (MSIDThrottlingType)processErrorResponseToGetThrottleType:(NSError *)errorResponse
+                                                      error:(NSError *_Nullable *_Nullable)error
+{
+    
+    MSIDThrottlingType throttleType = MSIDThrottlingTypeNone;
+    if ([MSIDThrottlingModel429 isApplicableForTheThrottleModel:errorResponse])
+    {
+        throttleType = MSIDThrottlingType429;
+        return throttleType;
+    }
+    
+    if ([MSIDThrottlingModelInteractionRequire isApplicableForTheThrottleModel:errorResponse])
+    {
+        throttleType = MSIDThrottlingTypeInteractiveRequired;
+        return throttleType;
+    }
+    
+    return throttleType;
+}
+
+
++ (BOOL)validateInput:(id<MSIDThumbprintCalculatable> _Nonnull)request
+{
+    return (request.fullRequestThumbprint || request.strictRequestThumbprint);
+}
+
+
++ (MSIDThrottlingCacheRecord *)getDBRecordWithStrictThumbprint:(NSString *)strictThumbprint
+                                                fullThumbprint:(NSString *)fullThumbprint
+                                                         error:(NSError **)error
+{
+    MSIDThrottlingCacheRecord *cacheRecord = [self.cacheService objectForKey:strictThumbprint
+                                                                       error:error];
+    if (!cacheRecord)
+    {
+        cacheRecord = [self.cacheService objectForKey:fullThumbprint error:error];
+    }
+    return cacheRecord;
+}
+
++ (MSIDLRUCache *)cacheService
+{
+    static MSIDLRUCache *cacheService = nil;
+    static dispatch_once_t onceToken;
+    dispatch_once(&onceToken, ^{
+        cacheService = [[MSIDLRUCache alloc] initWithCacheSize:1000];
```

Comment:
seems there's a duplicate instance of cache service between here and throttling model?

Reply:
Yes I realized it and the fix will be the next commit